catalog/from-k8s/resource.go

```diff
@@ -157,6 +161,13 @@ func (t *ServiceResource) shouldSync(svc *apiv1.Service) bool {
 		return false
 	}
 
+	// Ignore ClusterIP services if ClusterIP sync is disabled
+	if svc.Spec.Type == apiv1.ServiceTypeClusterIP {
+		if !t.ClusterIPSync {
```

Comment:
nit: I'd probably make this `if svc.Spec.Type == apiv1.ServiceTypeClusterIP &&  !t.ClusterIPSync {`